File: src/main/java/no/novari/flyt/integration/validation/constraints/ReferencedConfigurationExists.java
package no.novari.flyt.integration.validation.constraints;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Constraint(validatedBy = ReferencedConfigurationExistsValidator.class)
public @interface ReferencedConfigurationExists {

    String message() default "referenced configuration could not be found";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

}

File: src/main/kotlin/no/novari/flyt/integration/validation/constraints/ReferencedConfigurationExists.kt
package no.novari.flyt.integration.validation.constraints

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [ReferencedConfigurationExistsValidator::class])
annotation class ReferencedConfigurationExists(
    val message: String = "referenced configuration could not be found",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
)
